--- origsrc/flexdll/reloc.ml	2009-08-06 14:18:31.831464100 -0500
+++ src/flexdll/reloc.ml	2009-08-06 14:18:52.125524200 -0500
@@ -942,7 +942,7 @@ let setup_toolchain () =
 	    "/lib/w32api";
 	    gcclib () ];
       default_libs := ["-lkernel32"; "-luser32"; "-ladvapi32";
-		       "-lshell32"; "-lcygwin"; "-lgcc"]
+		       "-lshell32"; "-lcygwin"; "-lgcc_s"]
   | `MSVC ->
       search_path := !dirs @
 	parse_libpath (try Sys.getenv "LIB" with Not_found -> "");
