--- origsrc/flexdll/reloc.ml	2010-09-12 01:32:49.761088900 -0500
+++ src/flexdll/reloc.ml	2010-09-12 22:35:50.736809400 -0500
@@ -985,7 +985,7 @@ let setup_toolchain () =
 	    "/lib/w32api";
 	    gcclib () ];
       default_libs := ["-lkernel32"; "-luser32"; "-ladvapi32";
-		       "-lshell32"; "-lcygwin"; "-lgcc"]
+		       "-lshell32"; "-lcygwin"; "-lgcc_s"; "-lgcc"]
   | `MSVC | `MSVC64 ->
       search_path := !dirs @
 	parse_libpath (try Sys.getenv "LIB" with Not_found -> "");
