--- origsrc/flexdll-0.35/reloc.ml	2017-03-27 14:54:18.892826200 -0500
+++ src/flexdll-0.35/reloc.ml	2017-03-27 15:02:40.196472500 -0500
@@ -632,7 +632,10 @@ let build_dll link_exe output_file files
 
       (* Collect defined symbols *)
       List.iter
-        (fun sym -> if Symbol.is_defin sym then add_def sym.sym_name)
+        (fun sym -> if Symbol.is_defin sym
+                    then match check_prefix "__nm_" sym.sym_name with
+                      | Some s -> add_def s;
+                      | _ -> add_def sym.sym_name)
         obj.symbols
 
     and collect_file fn =
@@ -782,6 +785,11 @@ let build_dll link_exe output_file files
               Printf.printf "%s -> %s(%s) because of %s\n%!" fn libname objname s;
             link_libobj o
           with Not_found ->
+            try let (libname, objname, _) as o = defined_in ("__nm_" ^ s) in
+              if !explain then
+                Printf.printf "%s -> %s(%s) because of %s\n%!" fn libname objname s;
+              link_libobj o
+          with Not_found ->
             if !explain then
               Printf.printf "%s needs %s (not found)\n%!" fn s
       )
